models/eprop_iaf_psc_delta_adapt.h

```diff
@@ -0,0 +1,571 @@
+  both models are integrate-and-fire neurons with delta-shaped postsynaptic
+  currents, there are minor differences in the dynamics of the two models, such
+  as the propagator of the postsynaptic current and the voltage reset upon a
+  spike.
+
+E-prop plasticity was originally introduced and implemented in TensorFlow in [1]_.
+
+The membrane voltage time course is given by:
+
+.. math::
+    v_j^t &= \alpha v_j^{t-1}+\sum_{i \neq j}W_{ji}^\mathrm{rec}z_i^{t-1}
+             + \sum_i W_{ji}^\mathrm{in}x_i^t-z_j^{t-1}v_\mathrm{th} \,, \\
+    \alpha &= e^{-\frac{\delta t}{\tau_\mathrm{m}}} \,.
+
+The threshold adaptation is given by:
+
+.. math::
+    A_j^t &= v_\mathrm{th} + \beta a_j^t \,, \\
+    a_j^t &= \rho a_j^{t-1} + z_j^{t-1} \,, \\
+    \rho &= e^{-\frac{\delta t}{\tau_\mathrm{a}}} \,.
+
+The spike state variable is expressed by a Heaviside function:
+
+.. math::
+    z_j^t = H\left(v_j^t-A_j^t\right) \,.
+
+If the membrane voltage crosses the adaptive threshold voltage :math:`A_j^t`, a spike is
+emitted and the membrane voltage is reduced by :math:`v_\text{th}` in the next
+time step. After the time step of the spike emission, the neuron is not
+able to spike for an absolute refractory period :math:`t_\text{ref}`.
+
+An additional state variable and the corresponding differential equation
+represents a piecewise constant external current.
+
+Furthermore, the pseudo derivative of the membrane voltage needed for e-prop
+plasticity is calculated:
+
+.. math::
+    \psi_j^t = \frac{\gamma}{v_\text{th}} \text{max}
+               \left(0, 1-\left| \frac{v_j^t-A_j^t}{v_\text{th}}\right| \right) \,.
+
+See the documentation on the ``iaf_psc_delta`` neuron model for more information
+on the integration of the subthreshold dynamics.
+
+The change of the synaptic weight is calculated from the gradient
+:math:`\frac{\mathrm{d}{E}}{\mathrm{d}{W_{ij}}}=g`
+which depends on the presynaptic
+spikes :math:`z_i^{t-1}`, the surrogate-gradient / pseudo-derivative of the postsynaptic membrane
+voltage :math:`\psi_j^t` (which together form the eligibility trace
+:math:`e_{ji}`), and the learning signal :math:`L_j^t` emitted by the readout
+neurons.
+
+.. math::
+  \frac{\mathrm{d}E}{\mathrm{d}W_{ji}} = g &= \sum_t L_j^t \bar{e}_{ji}^t, \\
+  e_{ji}^t &= \psi_j^t \left(\bar{z}_i^{t-1} - \beta \epsilon_{ji,a}^{t-1}\right)\,, \\
+  \epsilon^{t-1}_{ji,\text{a}} &= \psi_j^{t-1}\bar{z}_i^{t-2} + \left( \rho - \psi_j^{t-1} \beta \right)
+  \epsilon^{t-2}_{ji,a}\; \text{with}\,\rho &= \exp\left(-\frac{\delta t}{\tau_\text{a}}\right)\,. \\
+
+The eligibility trace and the presynaptic spike trains are low-pass filtered
+with some exponential kernels:
+
+.. math::
+  \bar{e}_{ji}=\mathcal{F}_\kappa(e_{ji}) \;\text{with}\, \kappa=\exp\left(\frac{-\delta t}{\tau_\text{m,
+out}}\right)\,,\\ \bar{z}_i=\mathcal{F}_\alpha(z_i) \;\text{with}\, \alpha=\exp\left(\frac{-\delta
+t}{\tau_\text{m}}\right)\,.
```

Comment:
```suggestion
    t}{\tau_\text{m}}\right)\,.
```

This math equation fails to render and I think it's the indentation that's the reason